app/controllers/posts_controller.rb

```diff
@@ -25,9 +25,19 @@ def edit
   # POST /posts
   def create
     @post = Post.new(post_params)
-
     if @post.save
-      redirect_to @post, notice: 'Post was successfully created.'
+      tag_params.split(",").each do |tag_new|
```

Comment:
Po pierwsze - w kontrolerze nie powinno byc tyle kodu.
Kontroler sluzy do handlowania requestow. Ma dostac parametry i wyrenderowac odpowiedz/przekierowac usera gdzies.
Wszystko inne co sie musi stac zeby to zrobic powinno byc poza kotrolerem.
Normalnie tworzy sie osobna klase do tego, ale narazie uzyjmy istniejacych klas.

Po pierwsze dodaj do klasy `Tag` metode, ktora bedzie przyjmowac tagi z parametrow i dzielic je na tablice:
```ruby
def self.tags_from_params(tag_string)
# do work, return array
end
```

Zdefiniuj `","` jako stala w tej klasie i uzyj jej w metodzie `split`.

Uzyj zdefiniowanej metody w kontrolerze, przekazujac jej wynik do kolejnej metody, ktora zdefiniuj tez w klasie `Tag`:

```ruby
def self.find_or_create(tag_names)
  # przeiteruj po tagach, sprawdz czy sa w bazie, jak nie to stworz tag
  # zwroc tablice wszystkich tagow (obiektow, nie nazw)
end
```
Uzyj wyniku tej metody zeby podpiac tagi do posta, definiujac kolejna metode, tym razem w klasie `Post`.

Czyli w skrocie chcialabym zeby zapisywanie tagow tutaj wygladalo tak:

```ruby

  tag_names = Tag.tags_from_params(params[:post_tags])
  tags = Tag.find_or_create(tag_names)
  Post.assign_tags(tags)

```